### PR TITLE
Add nav menu

### PR DESCRIPTION
**Pull request by justafish**
Adds a navigation menu (with hardcoded links for now) and some (temporary) bad CSS so it's usable

ezgif-3-1c6e5fee5d

### CODE REVIEW

File: src/components/06_wrappers/Default/Default.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import { css } from 'emotion';
+import { scaleRotate as Menu } from 'react-burger-menu';
+import { Link } from 'react-router-dom';
+
+let styles;
+
+const menuLinks = {
+  home: [{ path: '/', text: '🏠 Home' }],
+  permissions: [{ path: '/admin/people/permissions', text: '🔐 Permissions' }],
+  content: [
```
Comment (dawehner) [inline src/components/06_wrappers/Default/Default.js]: I'm a bit confused that you are not using the same structure as the PR in https://github.com/jsdrupal/drupal-admin-ui/pull/23

File: src/components/05_pages/Permissions/Permissions.js [hunk 2]
```diff
@@ -19,12 +19,20 @@ const Permissions = class Permissions extends Component {
           `${
             process.env.REACT_APP_DRUPAL_BASE_URL
           }/admin-api/permissions?_format=json`,
+          {
```
Comment (dawehner) [inline src/components/05_pages/Permissions/Permissions.js]: Do you mind moving this into its own PR :)

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: I hadn't seen that yet, this is only supposed to be temporary as noted

Comment (dawehner) [reply src/components/06_wrappers/Default/Default.js]: Sure, it'd be nice to come up with a usecase based structure on the other PR though

**Comment by dawehner** [comment]
It'd be nice to not be too playfull here. Otherwise people start to nitpick on that.

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: I was going to wait until the other PR is committed and then create a new one to use the new endpoint

File: src/components/05_pages/Permissions/Permissions.js (hunk 2 above)
Comment (justafish) [reply src/components/05_pages/Permissions/Permissions.js]: Can we leave it for now? I was going to work on a follow up PR that puts all this API stuff into a separate wrapper file as there's a lot of repetition here

**Comment by justafish** [comment]
@dawehner styling is totally temporary, just wanted to get something in place so the menu is usable

**Comment by dawehner** [comment]
Well, just remove the crazy effect :)

**Comment by justafish** [comment]
nooo @dawehner that's my favourite part! 😂 I chose that one because it looks nice, shows off cool things we can do with components from npm, and doesn't require the page to be responsive (unlike the one which squashes stuff in, for example).
Also, just want to re-iterate that all the styling here is temporary! If this inspires people to come and nitpick on it and make something better than that's a +1 as far as I'm concerned 😁

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (dawehner) [reply src/components/06_wrappers/Default/Default.js]: Fair :) I also just asked @tedbow to maybe think about a better structure than what is right now in the other PR, so its good to not be blocked on this right now.

**Comment by dawehner** [comment]
Let's just use the crazy effect right now and then get the UX team actually involved. I think this effect shows that its potentially not the end of what we want :)

File: src/components/05_pages/Permissions/Permissions.js (hunk 2 above)
Comment (dawehner) [reply src/components/05_pages/Permissions/Permissions.js]: I'd suggest to not abstract till we have another usecase.

Comment (dawehner) [reply src/components/05_pages/Permissions/Permissions.js]: I would have suggest @mattgrill to work on it, given he has actually written a library doing exactly that.

Comment (dawehner) [reply src/components/05_pages/Permissions/Permissions.js]: To clarify, I wanted to express: @mattgrill would be a good candidate when you don't have time to work on it.

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (mattgrill) [reply src/components/06_wrappers/Default/Default.js]: @justafish @dawehner I mentioned that we should just merge the menu pr now so we can continue to quickly iterate on this.

File: src/components/05_pages/Permissions/Permissions.js (hunk 2 above)
Comment (mattgrill) [reply src/components/05_pages/Permissions/Permissions.js]: @justafish @dawehner I don't mind working on this less-than-glamorous part. I was already planning on doing something like this when I do the changes to update Drupal with the permissions changes.

Comment (mattgrill) [reply src/components/05_pages/Permissions/Permissions.js]: Additionally, I think this is fine to add here. I'll probably just copy all this out into it's own helper anyways.

**Comment by mattgrill** [comment]
One of the things we've tried in the past is to add a reduced animation mode. I think that having something like that, that you can configure would be nice. I'm not sure it's a blocker for this though. Maybe an issue/todo we can add.

File: src/App.js [hunk 3]
```diff
@@ -16,30 +17,15 @@ class App extends Component {
...
-          <hr />
-
-          <Route exact path="/" component={Home} />
-          {Object.keys(routes).map(route => (
-            <Route path={route} component={routes[route]} key={route} />
-          ))}
-          <Route component={NoMatch} />
-        </div>
+        <Default>
+          <Switch>
```
Comment (mattgrill) [inline src/App.js]: This is great!

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (mattgrill) [reply src/components/06_wrappers/Default/Default.js]: It might be worth it to add some sort of normalizer to the returned JSON from #23 to get to this point?

File: src/components/06_wrappers/Default/Default.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    </Menu>
+
+    <main className={styles.main} id={styles.main}>
+      {props.children}
+    </main>
+  </div>
+);
+
+Default.propTypes = {
+  children: PropTypes.node.isRequired,
```
Comment (mattgrill) [inline src/components/06_wrappers/Default/Default.js]: In `Permissions` I dropped the `PropTypes` bit and just used the exact element type. It comes out looking a bit cleaner.

File: src/components/06_wrappers/Default/Default.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+};
+
+styles = {
```
Comment (mattgrill) [inline src/components/06_wrappers/Default/Default.js]: Do we want to maybe but these in a different file?

Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: I like having them together - like JS and HTML being in the same file 😁

File: src/components/06_wrappers/Default/Default.js (hunk 4 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: oh, I'm surprised the linter let you do that! (given https://github.com/facebook/prop-types/issues/150)

Comment (mattgrill) [reply src/components/06_wrappers/Default/Default.js]: Yep. Works just fine.

File: src/components/06_wrappers/Default/Default.js (hunk 5 above)
Comment (mattgrill) [reply src/components/06_wrappers/Default/Default.js]: I can only see this getting more unmanageable as each of these files gets more complicated.

File: src/components/06_wrappers/Default/Default.js (hunk 4 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: :+1:

File: src/components/06_wrappers/Default/Default.js (hunk 5 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: @mattgrill we haven't had this issue when using it in the wild - generally they end up being split into smaller components if that's becoming the case. It's also much easier to work with all in the same place when you start using props inside your styles, styled components etc - much like not having a separate HTML and JS file anymore.

File: src/components/05_pages/Permissions/Permissions.js (hunk 2 above)
Comment (justafish) [reply src/components/05_pages/Permissions/Permissions.js]: :+1:

File: src/components/06_wrappers/Default/Default.js (hunk 1 above)
Comment (justafish) [reply src/components/06_wrappers/Default/Default.js]: :+1:

**Comment by justafish** [comment]
@mattgrill PR updated with destructured propTypes
